agave/chalice/rest_api.py

```
@@ -2,12 +2,19 @@
 from typing import Any, Optional, Type, cast
 from urllib.parse import urlencode
 
-from chalice import Blueprint, NotFoundError, Response
+try:
+    from chalice import Blueprint, NotFoundError, Response
+except ImportError:
+    raise ImportError(
+        "You must install agave with [chalice] option.\n"
+        "You can install it with: pip install agave[chalice]"
+    )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)